sdk/go/common/apitype/service.go

```diff
@@ -56,3 +54,41 @@ type APICapabilityConfig struct {
 type CapabilitiesResponse struct {
 	Capabilities []APICapabilityConfig `json:"capabilities"`
 }
+
+// Represents feature-detected capabilities of the service the backend is connected to.
+type Capabilities struct {
+	// If non-nil, indicates that delta checkpoint updates are supported.
+	DeltaCheckpointUpdates *DeltaCheckpointUploadsConfigV2
+
+	// Indicates whether the service supports bulk encryption.
+	BulkEncryption bool
+}
+
+func (r CapabilitiesResponse) Parse() (Capabilities, error) {
+	var parsed Capabilities
+	for _, entry := range r.Capabilities {
+		switch entry.Capability {
+		case DeltaCheckpointUploads:
+			var upcfg DeltaCheckpointUploadsConfigV2
+			if err := json.Unmarshal(entry.Configuration, &upcfg); err != nil {
+				msg := "decoding DeltaCheckpointUploadsConfig returned %w"
+				return Capabilities{}, fmt.Errorf(msg, err)
```

Comment:
Feel like it'd be neater to just inline the `msg :=` here and below (as in, just `return Capabilities{}, fmt.Errorf("decoding ... %w", ...)`)